ott/core/bar_problems.py

```diff
@@ -95,8 +95,14 @@ def segmented_y_b(
       return self.add_slice_for_debiased(self._y, self._b)
     else:
       segmented_y, segmented_b, _ = segment.segment_point_cloud(
-          self._y, self._b, self._segment_ids, self._num_segments,
-          self._indices_are_sorted, self._num_per_segment, self.max_measure_size
+          self._y,
+          self._b,
+          self._segment_ids,
+          self._num_segments,
+          self._indices_are_sorted,
+          self._num_per_segment,
+          self.max_measure_size,
+          padding_vector=self.cost_fn.padding_vector
```

Comment:
rather than passing a `padding_vector` "function" to `segment_point_cloud`, I think it would be easier to read to pass the `padding_vector` itself (what you called `padding` in the `segment_point_cloud` function).

The rationale for this is that `bar_problem` already has the knowledge of the `cost_fn` and `segment_point_cloud` is very low level, so it makes more sense that it only takes `jnp.ndarray` and `float`/`int` as inputs.

Reply:
Yes, I see your point. 

The only issue I see with passing the padding vector directly ( and not calling the method of the cost class) is that in the case where the segmentation is not happening for a barycenter problem but, for example in order to compute a lot of sinkhorn divergences between measures of different size) the user would have to be trusted to pass the appropriate padding vector. This would hold both for ``segment_point_cloud`` and ``segment_sinkhorn_divergence``